src/ShrinkCompilerServiceProvider.php

```diff
@@ -16,9 +17,9 @@ class ShrinkCompilerServiceProvider extends ServiceProvider
     public function register()
     {
         $this->app->singleton('blade.compiler', function () {
-            return new ShrinkCompiler(
-                $this->app['files'], $this->app['config']['view.compiled']
-            );
+            return tap(new ShrinkCompiler($this->app['files'], $this->app['config']['view.compiled']), function ($blade) {
+                $blade->component('dynamic-component', DynamicComponent::class);
```

Comment:
I realized the `\Illuminate\View\DynamicComponent` was added from commit [f66c103a](https://github.com/laravel/framework/commit/f66c103a9457328f0984a010c27afdf889b3c562) into branch `8.x`.
Which means if we want to keep the support for Laravel 6.x after new version be released, we must distinguish between 6.x and 8.x blade complier's constructor. The simple method is test if `class_exists('Illuminate\View\DynamicComponent') == true`:
- it's true then construct as new (8.x)
- otherwise do as old (6.x)

@HDVinnie PTAL

Reply:
every time when Laravel app start run , `class_exists()` may cost negligible performance, we also could remove the support for Laravel `6.x` after `"laravel-view-shrinker:8.0.0"`.

Reply:
Honestly I would just support the latest Laravel and drop v6 support

Reply:
sounds great